polkadot/xcm/xcm-builder/src/tests/barriers.rs

```diff
@@ -309,3 +309,62 @@ fn suspension_should_work() {
 	);
 	assert_eq!(r, Ok(()));
 }
+
+#[test]
+fn allow_subscriptions_from_should_work() {
+	// allow only parent
+	AllowSubsFrom::set(vec![Location::parent()]);
+
+	let valid_xcm_1 = Xcm::<TestCall>(vec![SubscribeVersion {
+		query_id: 42,
+		max_response_weight: Weight::from_parts(5000, 5000),
+	}]);
+	let valid_xcm_2 = Xcm::<TestCall>(vec![UnsubscribeVersion]);
+	let invalid_xcm_1 = Xcm::<TestCall>(vec![
+		SetAppendix(Xcm(vec![])),
+		SubscribeVersion { query_id: 42, max_response_weight: Weight::from_parts(5000, 5000) },
+	]);
+	let invalid_xcm_2 = Xcm::<TestCall>(vec![
+		SubscribeVersion { query_id: 42, max_response_weight: Weight::from_parts(5000, 5000) },
+		SetTopic([0; 32]),
+	]);
+
+	let test_data = vec![
```

Comment:
Nit: Not very important, but personally I would find it easier to follow and I think it would also be more extensible if we just asserted each case separately instead of iterating through a `vec`. I don't think it would create a lot of duplication.